app/src/main/java/fr/free/nrw/commons/Achievements.java

```diff
@@ -0,0 +1,104 @@
+
+    private static final double badge_image_ratio_width = 0.4;
+    private static final double badge_image_ratio_height = 0.36;
+    private ImageView imageView;
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_achievements);
+
+        DisplayMetrics displayMetrics = new DisplayMetrics();
+        getWindowManager().getDefaultDisplay().getMetrics(displayMetrics);
+        int height = displayMetrics.heightPixels;
+        int width = displayMetrics.widthPixels;
+
+        imageView = (ImageView)findViewById(R.id.achievement_badge);
+
+        RelativeLayout.LayoutParams params = (RelativeLayout.LayoutParams)
+                imageView.getLayoutParams();
+        params.height = (int) (height*badge_image_ratio_height);
+        params.width = (int) (width*badge_image_ratio_width);
+        imageView.setImageResource(R.drawable.sydney_opera_house);
+        imageView.requestLayout();
+
+        android.support.v7.widget.Toolbar toolbar = (android.support.v7.widget.Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        getMenuInflater().inflate(R.menu.menu_about, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        int id = item.getItemId();
+        if (id == R.id.share_app_icon) {
+            View rootView = getWindow().getDecorView().findViewById(android.R.id.content);
+            Bitmap screenShot = getScreenShot(rootView);
+            imageView.setImageBitmap(screenShot);
+            shareScreen(screenShot);
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+    public static Bitmap getScreenShot(View view) {
+        View screenView = view.getRootView();
+        screenView.setDrawingCacheEnabled(true);
+        Bitmap bitmap = Bitmap.createBitmap(screenView.getDrawingCache());
+        screenView.setDrawingCacheEnabled(false);
+        return bitmap;
+    }
+
+    void shareScreen ( Bitmap bitmap){
+        try {
+            File file = new File(this.getExternalCacheDir(),"screen.png");
+            FileOutputStream fOut = new FileOutputStream(file);
+            bitmap.compress(Bitmap.CompressFormat.PNG, 100, fOut);
+            fOut.flush();
+            fOut.close();
+            file.setReadable(true, false);
+            final Intent intent = new Intent(android.content.Intent.ACTION_SEND);
+            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+            intent.putExtra(Intent.EXTRA_STREAM, Uri.fromFile(file));
+            intent.setType("image/png");
+            startActivity(Intent.createChooser(intent, "Share image via"));
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Do we really need to catch all exceptions? IMO `IOException` should be sufficient.